File: data/zzzIncluder.java
import java.io.File;
import java.io.FileNotFoundException;
import java.io.FileWriter;
import java.io.IOException;
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class zzzIncluder {

	public static void main(String[] args) throws IOException {
		List<String> list = include(args[0]);
		FileWriter writer = new FileWriter(args[1]);
		for (String s : list) {
			writer.write(s + "\n");
		}
		writer.close();
	}

	public static List<String> include(String fileName) throws FileNotFoundException {
		List<String> file = new ArrayList();
		Scanner scan = new Scanner(new File(fileName));
		List<String> ret = new ArrayList();
		while(scan.hasNextLine()) {
			file.add(scan.nextLine());
		}
		for (int i = 0; i < file.size(); i++) {
			if (file.get(i).trim().length() >= 8 && file.get(i).trim().substring(0, 8).equals("#include")) {
				String name = file.get(i).trim();
				name = name.substring(name.indexOf("\"") + 1, name.length() - 1);
				List<String> included = include(name);
				for (String s : included) {
					ret.add(s);
				}
			} else {
				ret.add(file.get(i));
			}
		}
		return ret;
	}
}

File: data/zzzIncluder.cpp
#include <fstream>
#include <iostream>
#include <vector>

using namespace std;

vector<string>* include(string file_name) {
	vector<string> file;
	ifstream in(file_name);
	string line;
	vector<string>* ret = new vector<string>();
	while (getline(in, line)) {
		file.push_back(line);
	}
	for (int i = 0; i < file.size(); i++) {
		if (file[i].length() >= 8 && file[i].substr(0, 8) == "#include") {
			string name = file[i].substr(9, string::npos);
			name = name.substr(1, name.length() - 2);
			vector<string>* included_p = include(name);
			vector<string> included = *included_p;
			delete included_p;
			for (string s : included) {
				ret->push_back(s);
			}
		} else {
			ret->push_back(file[i]);
		}
	}
	return ret;
}

int main(int argc, char* args[]) {
	vector<string> v = *include(args[1]);
	ofstream writer(args[2]);
	for (string s : v) {
		writer << s << endl;
	}
	writer.close();
}